distributed/core.py

```diff
@@ -238,8 +237,6 @@ def set_thread_ident():
 
         self.__stopped = False
 
-        super().__init__(**kwargs)
-
```

Comment:
These were needed because of [the inheritance in `Scheduler`]( https://github.com/dask/distributed/blob/233ec8840671b5d705a06d0dcfe1ff42b771230b/distributed/scheduler.py#L3176 ) ( https://github.com/dask/distributed/pull/4365#discussion_r557630488 ). Is there some reason to remove them? Would suspect they are still needed

Reply:
See https://github.com/dask/distributed/pull/4733#discussion_r618547785 I believe this is only needed for classes which are sandwiched, i.e. are the intermediates in the diamond inheritance. This class is always root and therefore will not need it. In fact, calling super here *with* arguments will raise an exception since _super_ is `object` which cannot be initialized with arguments

Reply:
I think this was added to make us more resilient to the inheritance order (see [this earlier comment](https://github.com/dask/distributed/pull/4733#discussion_r618603420) and @gjoseph92's nice write up over in https://github.com/dask/distributed/issues/4640). One thing this PR is proposing is that `Server` is always at the bottom of the mro for any classes which inherit from `Server`. This is already the case today and we could add a test to make sure continues to be true in the future.

Reply:
In this specific case, we need it for `SchedulerState` and `ServerNode` but not for `Server`. Our case is actually even a tiny bit simpler since we do not have a diamond

```
Scheduler -> SchedulerState -> object
Scheduler -> ServerNode -> Server -> object
```

with mro


```python
In [1]: from distributed import Scheduler

In [2]: Scheduler.mro()
Out[2]:
[distributed.scheduler.Scheduler,
 distributed.scheduler.SchedulerState,
 distributed.node.ServerNode,
 distributed.core.Server,
 object]
```

Reply:
Oops sorry for the 2nd thread 😬

Unfortunately with a Cythonized class I think we need it first as opposed to second. Trying the latter resulted in compilation errors. That said, one of the things I've discussed with Matt before is refactoring `SchedulerState` into its own object (to make inheritance unneeded), which should be useful for a few reasons. Though this is a bit more involved change so may not happen right away and could uncover other issues

Reply:
> Unfortunately with a Cythonized class I think we need it first as opposed to second. Trying the latter resulted in compilation errors

Sorry, I don't quite follow this point. Given that we're cythonizing the scheduler in some CI builds, I suspect the current class structure we have today is okay (though I could be missing something)

Reply:
All I meant was `SchedulerState` needs to have higher precedence. I don't think anything here is changing that (was just unclear on whether changes to this order were planned)

IOW the code as changed seem ok to me